Userland/Libraries/LibCrypto/BigFraction/BigFraction.cpp

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (c) 2022, Lucas Chollet <lucas.chollet@free.fr>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "BigFraction.h"
+#include <AK/Math.h>
+#include <AK/String.h>
+#include <AK/StringBuilder.h>
+#include <LibCrypto/NumberTheory/ModularFunctions.h>
+
+namespace Crypto {
+
+BigFraction::BigFraction(Crypto::SignedBigInteger numerator, Crypto::UnsignedBigInteger denominator)
+    : m_numerator(move(numerator))
+    , m_denominator(move(denominator))
+{
+    VERIFY(m_denominator != 0);
+    reduce();
+}
+
+BigFraction::BigFraction(Crypto::SignedBigInteger value)
+    : BigFraction(move(value), 1)
+{
+}
+
+BigFraction::BigFraction(StringView sv)
```

Comment:
This very much looks like it should be a fallible API

Reply:
I do understand your concern, but for now it is only used in a lambda (with no error propagation) and `SignedBigInteger::from_base` doesn't propagate error either. And finally, all the API uses in place `VERIFY()` instead of propagating error due to the massive usage of operators.

Is it better to duplicate all the methods with fallible and non-fallible functions, only incorporate a bunch of `VERIFY` in this function, or just leave it as a `TODO` to not add too much in this PR?

Reply:
okay then leave a fixme